### PR TITLE
HW #15 "Maven/Gradle"

### PR DESCRIPTION
**Pull request by sviat-h**

### CODE REVIEW

File: src/main/java/hw15/controller/AppExecutor.java [hunk 1]
```diff
@@ -0,0 +1,21 @@
+package hw15.controller;
+
+import hw15.model.FileManager;
+
+import java.io.IOException;
+
+public class AppExecutor {
+    private FileManager fileManager = new FileManager();
+
+    public  void runProgram() {
```
Comment (omilovskiy) [inline src/main/java/hw15/controller/AppExecutor.java]: remove whitespace

File: src/main/java/hw15/controller/AppExecutor.java [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+        try {
+            fileManager.countWords();
```
Comment (omilovskiy) [inline src/main/java/hw15/controller/AppExecutor.java:13]: split each line of code with empty line

File: src/main/java/hw15/model/FileManager.java [hunk 3]
```diff
@@ -0,0 +1,104 @@
+package hw15.model;
+
+import java.io.*;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FileManager {
+    private String fileReadName = "src/main/resources/data.txt";
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: make a constant

File: src/main/java/hw15/model/FileManager.java [hunk 4]
```diff
@@ -0,0 +1,104 @@
...
+    private String newFileName = "src/main/resources/writeText.txt";
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: as previous

File: src/main/java/hw15/model/FileManager.java [hunk 5]
```diff
@@ -0,0 +1,104 @@
...
+
+    public void countWords() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+        String line;
+        int numberOfWords = 0;
+        while ((line = bufferedReader.readLine()) != null) {
+            songArray = line.split(" ");
+            numberOfWords = numberOfWords + songArray.length;
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: numberOfWords += songArray.length;

File: src/main/java/hw15/model/FileManager.java [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FileManager {
+    private String fileReadName = "src/main/resources/data.txt";
+    private String newFileName = "src/main/resources/writeText.txt";
+    private File file = new File(fileReadName);
+    private String[] songArray;
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: wordsArray

File: src/main/java/hw15/model/FileManager.java [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+            numberOfWords = numberOfWords + songArray.length;
+        }
+        System.out.println("The quantity of all words in the song :" + numberOfWords);
+        bufferedReader.close();
+    }
+
+    public void writeShortWordsToNewFile() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+        FileWriter fileWriter = new FileWriter(newFileName);
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: add empty line above

File: src/main/java/hw15/model/FileManager.java [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+        BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+
+        String line;
+        int numberOfWords = 0;
+        while ((line = bufferedReader.readLine()) != null) {
+            String[] shortWordsArray = line.split(" ");
+
+            for (String s : shortWordsArray) {
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: String shortWord

File: src/main/java/hw15/model/FileManager.java [hunk 9]
```diff
@@ -0,0 +1,104 @@
...
+                    bufferedWriter.write(s + "\n");
+                    bufferedWriter.flush();
+
+                    numberOfWords++;
+
+                }
+            }
+        }
+        System.out.println("The quantity of short words in the song :" + numberOfWords + "\n");
+        bufferedReader.close();
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:66]: add empty line above

File: src/main/java/hw15/model/FileManager.java [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+        bufferedWriter.close();
+    }
+
+    public void readShortWordsFromNewFile() throws IOException {
+        FileReader fileReader = new FileReader(newFileName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+        System.out.println("Read from a new file: ");
+        String line;
+        while ((line = bufferedReader.readLine()) != null) {
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:79]: as previous

File: src/main/java/hw15/model/FileManager.java [hunk 11]
```diff
@@ -0,0 +1,104 @@
...
+            System.out.print(line + ", ");
+        }
+        System.out.println("\n");
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:83]: as previous

File: src/main/java/hw15/model/FileManager.java [hunk 12]
```diff
@@ -0,0 +1,104 @@
...
+        bufferedReader.close();
+    }
+
+    public void writeAllWordsToList() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+        String line;
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:92]: as previous

File: src/main/java/hw15/model/FileManager.java [hunk 13]
```diff
@@ -0,0 +1,104 @@
...
+
+        while ((line = bufferedReader.readLine()) != null) {
+            songArray = line.split(" ");
+            int i = 0;
+            for (String s : songArray) {
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: String word

File: src/main/java/hw15/model/FileManager.java [hunk 14]
```diff
@@ -0,0 +1,104 @@
...
+                if (!s.equals("" + "")) {
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: what is this: "" + ""??

File: src/main/java/hw15/model/FileManager.java [hunk 15]
```diff
@@ -0,0 +1,104 @@
...
+                    listOfAllWords.add(i, s.toLowerCase());
+                }
+                i++;
+            }
+        }
+        bufferedReader.close();
+    }
+
+    public void showMostCommonWords() {
+        Map<String, Long> mostCommonWords = listOfAllWords.stream()
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: mostCommonWords -> wordToAmount

File: src/main/java/hw15/model/FileManager.java [hunk 16]
```diff
@@ -0,0 +1,104 @@
...
+                .collect(Collectors.groupingBy(listOfAllWords -> listOfAllWords, Collectors.counting()));
+
+        List<Map.Entry<String, Long>> collect = mostCommonWords.entrySet().stream()
+                .sorted(Map.Entry.<String, Long>comparingByValue().reversed())
+                .collect(Collectors.toList());
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: try to make Map with `NUMBER_OF_POPULAR_WORDS` amount
and y will  get rid of for loop lower

File: src/main/java/hw15/model/FileManager.java [hunk 17]
```diff
@@ -0,0 +1,104 @@
...
+    private static final int LETTERS_LIMIT = 3;
+
+    public void countWords() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+        String line;
+        int numberOfWords = 0;
+        while ((line = bufferedReader.readLine()) != null) {
+            songArray = line.split(" ");
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: " " -> move to constant

File: src/main/java/hw15/model/FileManager.java [hunk 18]
```diff
@@ -0,0 +1,104 @@
...
+
+    public void writeShortWordsToNewFile() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+        FileWriter fileWriter = new FileWriter(newFileName);
+        BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+
+        String line;
+        int numberOfWords = 0;
+        while ((line = bufferedReader.readLine()) != null) {
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:50]: as previous

File: src/main/java/hw15/model/FileManager.java [hunk 19]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    public void writeAllWordsToList() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+        String line;
+
+        while ((line = bufferedReader.readLine()) != null) {
+            songArray = line.split(" ");
+            int i = 0;
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: what is "i"?? it should be some counter

**Review by omilovskiy — COMMENTED** [review]
Hey:) Check my comments

File: src/main/java/hw15/controller/AppExecutor.java [hunk 20]
```diff
@@ -0,0 +1,21 @@
...
+            fileManager.writeShortWordsToNewFile();
+            fileManager.readShortWordsFromNewFile();
+            fileManager.writeAllWordsToList();
+            fileManager.showMostCommonWords();
+        } catch (IOException e) {
```
Comment (omilovskiy) [inline src/main/java/hw15/controller/AppExecutor.java:23]: add empty line above

File: src/main/java/hw15/model/FileManager.java (hunk 3 above)
Comment (omilovskiy) [reply src/main/java/hw15/model/FileManager.java]: pathToFile
rename

File: src/main/java/hw15/model/FileManager.java (hunk 4 above)
Comment (omilovskiy) [reply src/main/java/hw15/model/FileManager.java]: pathToDestinationFile
rename

File: src/main/java/hw15/model/FileManager.java [hunk 21]
```diff
@@ -0,0 +1,104 @@
...
+import java.util.Map;
+import java.util.stream.Collectors;
+
+public class FileManager {
+    private String fileReadName = "src/main/resources/data.txt";
+    private String newFileName = "src/main/resources/writeText.txt";
+    private File file = new File(fileReadName);
+    private String[] songArray;
+    private List<String> listOfAllWords = new ArrayList<>();
+    private static final int NUMBER_OF_POPULAR_WORDS = 10;
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:17]: add empty line above

File: src/main/java/hw15/model/FileManager.java [hunk 22]
```diff
@@ -0,0 +1,104 @@
...
+    private static final int LETTERS_LIMIT = 3;
+
+    public void countWords() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+        String line;
+        int numberOfWords = 0;
+        while ((line = bufferedReader.readLine()) != null) {
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:28]: as previous

File: src/main/java/hw15/model/FileManager.java [hunk 23]
```diff
@@ -0,0 +1,104 @@
...
+    private String newFileName = "src/main/resources/writeText.txt";
+    private File file = new File(fileReadName);
+    private String[] songArray;
+    private List<String> listOfAllWords = new ArrayList<>();
+    private static final int NUMBER_OF_POPULAR_WORDS = 10;
+    private static final int LETTERS_LIMIT = 3;
+
+    public void countWords() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:23]: as previous

File: src/main/java/hw15/model/FileManager.java [hunk 24]
```diff
@@ -0,0 +1,104 @@
...
+
+        String line;
+        int numberOfWords = 0;
+        while ((line = bufferedReader.readLine()) != null) {
+            songArray = line.split(" ");
+            numberOfWords = numberOfWords + songArray.length;
+        }
+        System.out.println("The quantity of all words in the song :" + numberOfWords);
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:33]: add empty line above

File: src/main/java/hw15/model/FileManager.java [hunk 25]
```diff
@@ -0,0 +1,104 @@
...
+        bufferedReader.close();
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:35]: add empty line above

File: src/main/java/hw15/model/FileManager.java [hunk 26]
```diff
@@ -0,0 +1,104 @@
...
+    }
+
+    public void writeShortWordsToNewFile() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:41]: add empty line above

File: src/main/java/hw15/model/FileManager.java [hunk 27]
```diff
@@ -0,0 +1,104 @@
...
+        FileWriter fileWriter = new FileWriter(newFileName);
+        BufferedWriter bufferedWriter = new BufferedWriter(fileWriter);
+
+        String line;
+        int numberOfWords = 0;
+        while ((line = bufferedReader.readLine()) != null) {
+            String[] shortWordsArray = line.split(" ");
+
+            for (String s : shortWordsArray) {
+                if (s.length() <= LETTERS_LIMIT && !s.equals("" + "")) {
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: what is: "" + ""??

File: src/main/java/hw15/model/FileManager.java [hunk 28]
```diff
@@ -0,0 +1,104 @@
...
+                    bufferedWriter.write(s + "\n");
+                    bufferedWriter.flush();
+
+                    numberOfWords++;
+
+                }
+            }
+        }
+        System.out.println("The quantity of short words in the song :" + numberOfWords + "\n");
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:64]: add empty line above

File: src/main/java/hw15/model/FileManager.java [hunk 29]
```diff
@@ -0,0 +1,104 @@
...
+        bufferedReader.close();
+        bufferedWriter.close();
+    }
+
+    public void readShortWordsFromNewFile() throws IOException {
+        FileReader fileReader = new FileReader(newFileName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
+
+        System.out.println("Read from a new file: ");
+        String line;
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:77]: add empty line above

File: src/main/java/hw15/model/FileManager.java [hunk 30]
```diff
@@ -0,0 +1,104 @@
...
+            }
+        }
+        System.out.println("The quantity of short words in the song :" + numberOfWords + "\n");
+        bufferedReader.close();
+        bufferedWriter.close();
+    }
+
+    public void readShortWordsFromNewFile() throws IOException {
+        FileReader fileReader = new FileReader(newFileName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:73]: add empty line above

File: src/main/java/hw15/model/FileManager.java [hunk 31]
```diff
@@ -0,0 +1,104 @@
...
+        while ((line = bufferedReader.readLine()) != null) {
+            System.out.print(line + ", ");
+        }
+        System.out.println("\n");
+        bufferedReader.close();
+    }
+
+    public void writeAllWordsToList() throws IOException {
+        FileReader fileReader = new FileReader(fileReadName);
+        BufferedReader bufferedReader = new BufferedReader(fileReader);
```
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:90]: add empty line above

File: src/main/java/hw15/model/FileManager.java (hunk 12 above)
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java:92]: add empty line above

File: src/main/java/hw15/model/FileManager.java (hunk 19 above)
Comment (omilovskiy) [inline src/main/java/hw15/model/FileManager.java]: add empty line above

**Review by omilovskiy — COMMENTED** [review]
Pls, check my comments

File: src/main/java/hw15/controller/AppExecutor.java (hunk 1 above)
Comment (sviat-h) [reply src/main/java/hw15/controller/AppExecutor.java]: +

File: src/main/java/hw15/controller/AppExecutor.java (hunk 2 above)
Comment (sviat-h) [reply src/main/java/hw15/controller/AppExecutor.java:13]: +

File: src/main/java/hw15/controller/AppExecutor.java (hunk 20 above)
Comment (sviat-h) [reply src/main/java/hw15/controller/AppExecutor.java:23]: +

File: src/main/java/hw15/model/FileManager.java (hunk 3 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 4 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 6 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 21 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:17]: +

File: src/main/java/hw15/model/FileManager.java (hunk 23 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:23]: +

File: src/main/java/hw15/model/FileManager.java (hunk 22 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:28]: +

File: src/main/java/hw15/model/FileManager.java (hunk 17 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 5 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 24 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:33]: +

File: src/main/java/hw15/model/FileManager.java (hunk 25 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:35]: +

File: src/main/java/hw15/model/FileManager.java (hunk 26 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:41]: +

File: src/main/java/hw15/model/FileManager.java (hunk 7 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 18 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:50]: +

File: src/main/java/hw15/model/FileManager.java (hunk 8 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 28 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:64]: +

File: src/main/java/hw15/model/FileManager.java (hunk 9 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:66]: +

File: src/main/java/hw15/model/FileManager.java (hunk 30 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:73]: +

File: src/main/java/hw15/model/FileManager.java (hunk 29 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:77]: +

File: src/main/java/hw15/model/FileManager.java (hunk 10 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:79]: +

File: src/main/java/hw15/model/FileManager.java (hunk 11 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:83]: +

File: src/main/java/hw15/model/FileManager.java (hunk 31 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:90]: +

File: src/main/java/hw15/model/FileManager.java (hunk 12 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:92]: +

Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java:92]: +

File: src/main/java/hw15/model/FileManager.java (hunk 19 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 13 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 15 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 16 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 27 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

File: src/main/java/hw15/model/FileManager.java (hunk 14 above)
Comment (sviat-h) [reply src/main/java/hw15/model/FileManager.java]: +

**Review by omilovskiy — APPROVED** [review]

**Pull request merged**